api.planx.uk/modules/team/service.ts

```diff
@@ -0,0 +1,74 @@
+import { getClient } from "../../client";
+import { TeamRole } from "@opensystemslab/planx-core/types";
+
+interface UpsertMember {
+  userEmail: string;
+  teamSlug: string;
+  role: TeamRole;
+}
+
+interface RemoveUser {
+  userEmail: string;
+  teamSlug: string;
+}
+
+export const addMember = async ({
+  userEmail,
+  teamSlug,
+  role,
+}: UpsertMember) => {
+  const $client = getClient();
+  const { user, team } = await getUserAndTeam({ userEmail, teamSlug });
+
+  const isSuccess = await $client.team.addMember({
+    teamId: team.id,
+    userId: user.id,
+    role,
+  });
+  if (!isSuccess) throw Error(`Failed to assign ${userEmail} to ${teamSlug}`);
+};
+
+export const changeMemberRole = async ({
+  userEmail,
+  teamSlug,
+  role,
+}: UpsertMember) => {
+  const $client = getClient();
+  const { user, team } = await getUserAndTeam({ userEmail, teamSlug });
```

Comment:
Doing multiple round trips to the DB to get these details feels pretty awkward. If / when we expose this functionality into the UI it would be worth having `userId` and `teamId` upfront to make these requests.